src/amo/components/SponsoredAddonsShelf/index.js

```diff
@@ -29,17 +36,42 @@ type Props = {|
 
 export type InternalProps = {|
   ...Props,
+  _config: typeof config,
   _tracking: typeof tracking,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
   i18n: I18nType,
+  isLoading: boolean,
   resultsLoaded: boolean,
+  shelfData: SponsoredShelfType | null | void,
   shelves: { [shelfName: string]: Array<AddonType> | null },
 |};
 
 export class SponsoredAddonsShelfBase extends React.Component<InternalProps> {
   static defaultProps = {
+    _config: config,
     _tracking: tracking,
+    shelfData: undefined,
   };
 
+  useAdzerk = () => {
```

Comment:
I don't think we need an arrow function here, we could simply have `useAdzerk() {}`, no?

Reply:
You are correct. I think I just tend to use arrow functions all the time now. I'll change it, but I'm curious, what is the disadvantage to using one here?

Reply:
The arrow function unnecessarily binds `this` in this case, there is no disadvantage but there is no advantage either. We usually don't use arrow functions in classes, unless we have to.